src/features/device-logs/lib/read.ts

```diff
@@ -115,21 +115,30 @@ async function handleStreamingRead(
 		}
 	});
 
-	function heartbeat() {
-		if (state !== StreamState.Closed) {
-			// In order to keep the connection alive, output new lines every now and then
-			write('\n');
-			setTimeout(heartbeat, LOGS_HEARTBEAT_INTERVAL);
-		}
-	}
-
-	setTimeout(heartbeat, LOGS_HEARTBEAT_INTERVAL);
+	let heartbeatInterval: ReturnType<typeof setInterval> | null = setInterval(
```

Comment:
If you set the empty case as `undefined` then you can just keep `clearInterval(heartbeatInterval);` as it works accepts `undefined` (it just doesn't like `null`)
```suggestion
	let heartbeatInterval: ReturnType<typeof setInterval> | undefined = setInterval(
```